static/app/views/settings/organizationMembers/organizationMemberRow.tsx

```diff
@@ -212,7 +213,7 @@ export default class OrganizationMemberRow extends PureComponent<Props, State> {
               </Confirm>
             )}
 
-            {showLeaveButton && !memberCanLeave && (
+            {showLeaveButton && !memberCanLeave && !flags['idp:provisioned'] && (
```

Comment:
It makes the code rather brittle to be checking for the IdP flag at this place. 

You should rework the logic.